packages/vow/src/watch.js

```diff
@@ -211,17 +212,18 @@ export const prepareWatch = (
 
     const { promiseWatcher, vowSetter } = makeWatcherKit(settler, watcher);
 
-    // Ensure we have a presence that won't be disconnected later.
-    unwrapPromise(specimenP, (specimen, payload) => {
-      vowSetter.setVow(specimen);
-      // Persistently watch the specimen.
-      if (!payload) {
-        // Specimen is not a vow.
-        promiseWatcher.onFulfilled(specimen);
-        return;
-      }
-      watchVow(specimen, promiseWatcher);
-    }).catch(e => promiseWatcher.onRejected(e));
+    if (getVowPayload(specimenP)) {
+      vowSetter.setVow(specimenP);
+    } else {
+      // TODO: Eat less of our own tail.
+      const {
+        settler: { resolve, reject },
```

Comment:
Destructuring methods from a settler will not work.  They need to be called as `settler.resolve`or `settler.reject` to preserve the `this` binding.